### PR TITLE
Fix apache-commons-lang and junit dependencies.

### PR DESCRIPTION
**Pull request by hrstoyanov**
This is a minor dependency clean up for stuff that need not end up in the Earrai app war file.

### CODE REVIEW

**Comment by tiagobento** [comment]
ok to test

**Comment by hrstoyanov** [comment]
@tiagobento @mbarkley 
Can you help me understand which test failed? The ```Details``` link does not work.

I am battling a failing CodegenTest that does not work on Windows 7 or MacOS. Moist likely the reason is that I have Java 9 and Java 8 installed, the test is launching separate process that somehow seems to grab the wrong java and fails all the time. Untilli figure out why this is happening, just let me know what fails in the automatic checks

**Comment by mbarkley** [comment]
@hrstoyanov Are you not receiving an email when the test fails? IIRC the email receives a public link that should be accessible to you.

**Comment by hrstoyanov** [comment]
@mbarkley 
I did receive email on Dec 7 for a prior pull request. Nothing for this one? And not even in my spam folder.

**Comment by hrstoyanov** [comment]
ok to test

**Comment by hrstoyanov** [comment]
Can someone take a look at why tests fail? I ran the pull request branch in a clean  OpenJDK 8 docker container and all tests passed...

**Comment by tiagobento** [comment]
Jenkins, please retest this

**Comment by hrstoyanov** [comment]
@tiagobento thanks, if it fails again, pls tell me where

**Comment by hrstoyanov** [comment]
Maven war plugin fails in a way that I can not figure out:

**mvn -e -s settings.xml install test** 
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-war-plugin:2.1.1:war (default-war) on project errai-bus-demo-stress-test: Failed to copy file for artifact [org.jboss.errai:errai-bus:jar:4.1.3-SNAPSHOT:compile]: /errai/err
ai-bus/target/classes (Is a directory) -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-war-plugin:2.1.1:war (default-war) on project errai-bus-demo-stress-test: Failed to copy file for artifact [org.jboss.errai:e
rrai-bus:jar:4.1.3-SNAPSHOT:compile]
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to copy file for artifact [org.jboss.errai:errai-bus:jar:4.1.3-SNAPSHOT:compile]
    at org.apache.maven.plugin.war.packaging.ArtifactsPackagingTask.performPackaging (ArtifactsPackagingTask.java:131)
    at org.apache.maven.plugin.war.packaging.WarProjectPackagingTask.handleArtifacts (WarProjectPackagingTask.java:189)
    at org.apache.maven.plugin.war.packaging.WarProjectPackagingTask.performPackaging (WarProjectPackagingTask.java:108)
    at org.apache.maven.plugin.war.AbstractWarMojo.buildWebapp (AbstractWarMojo.java:472)
    at org.apache.maven.plugin.war.AbstractWarMojo.buildExplodedWebapp (AbstractWarMojo.java:404)
    at org.apache.maven.plugin.war.WarMojo.performPackaging (WarMojo.java:197)
    at org.apache.maven.plugin.war.WarMojo.execute (WarMojo.java:159)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:134)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.io.FileNotFoundException: /errai/errai-bus/target/classes (Is a directory)
    at java.io.FileInputStream.open0 (Native Method)
    at java.io.FileInputStream.open (FileInputStream.java:195)
    at java.io.FileInputStream.<init> (FileInputStream.java:138)
    at org.codehaus.plexus.util.io.FileInputStreamFacade.getInputStream (FileInputStreamFacade.java:39)
    at org.codehaus.plexus.util.FileUtils.copyStreamToFile (FileUtils.java:1106)
    at org.codehaus.plexus.util.FileUtils.copyFile (FileUtils.java:1013)
    at org.apache.maven.plugin.war.packaging.AbstractWarPackagingTask.copyFile (AbstractWarPackagingTask.java:293)
    at org.apache.maven.plugin.war.packaging.AbstractWarPackagingTask$1.registered (AbstractWarPackagingTask.java:150)
    at org.apache.maven.plugin.war.util.WebappStructure.registerFile (WebappStructure.java:211)
    at org.apache.maven.plugin.war.packaging.AbstractWarPackagingTask.copyFile (AbstractWarPackagingTask.java:145)
    at org.apache.maven.plugin.war.packaging.ArtifactsPackagingTask.performPackaging (ArtifactsPackagingTask.java:106)
    at org.apache.maven.plugin.war.packaging.WarProjectPackagingTask.handleArtifacts (WarProjectPackagingTask.java:189)
    at org.apache.maven.plugin.war.packaging.WarProjectPackagingTask.performPackaging (WarProjectPackagingTask.java:108)
    at org.apache.maven.plugin.war.AbstractWarMojo.buildWebapp (AbstractWarMojo.java:472)
    at org.apache.maven.plugin.war.AbstractWarMojo.buildExplodedWebapp (AbstractWarMojo.java:404)
    at org.apache.maven.plugin.war.WarMojo.performPackaging (WarMojo.java:197)
    at org.apache.maven.plugin.war.WarMojo.execute (WarMojo.java:159)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:134)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
```

**Comment by mbarkley** [comment]
Not sure where that error you pasted comes from, but the problem on Jenkins appears to be a missing `org.apache.commons.lang3.StringUtils` in the Errai Bus Websocket Weld tests that causes the build to hang.

**Comment by hrstoyanov** [comment]
@mbarkley 
At a loss here ... running off my branch ```https://github.com/hrstoyanov/errai/tree/dependencies_cleanup``` produces no failed test. Also, if you look at the ```Errai:: Bus:JSR-356::WebSocket::Weld``` in the above branch, there is no reference to ```org.apache.commons.lang3.StringUtils``` anywhere in the unit tests at all?

Is there any chance Jenkins is pulling old patch and not the latest in the pull request?

**Comment by tiagobento** [comment]
Jenkins please retest this.

**Comment by hrstoyanov** [comment]
@tiagobento will take care of it next 1-2 days! As a remainder, the goal of this patch is to get rid of 3-4 jars with weak reasons to be used.

**Comment by kiereleaseuser2** [comment]
Can one of the admins verify this PR? Comment with 'ok to test' to start the build.

**Comment by hrstoyanov** [comment]
i still have not found the time to fix it, maybe this week

**Comment by kiereleaseuser2** [comment]
Build finished. No test results found.

**Comment by kiereleaseuser2** [comment]
Build finished. 187 tests run, 1 skipped, 6 failed.

**Comment by mareknovotny** [comment]
Jenkins retest this please

File: errai-common/pom.xml [hunk 1]
```diff
@@ -84,10 +85,6 @@
       <artifactId>gwt-slf4j</artifactId>
     </dependency>
 
-    <dependency>
-      <groupId>org.apache.commons</groupId>
-      <artifactId>commons-lang3</artifactId>
-    </dependency>
```
Comment (mareknovotny) [inline errai-common/pom.xml:88]: this dependency removal is not handled here in the PR correctly. There are multiple usages of commons-lang3 classes as transitive dependencies in errai-codegen or somewhere else too

**Review by mareknovotny — CHANGES_REQUESTED** [review]
There are compiler errors due removal of apache commons-lang3 dependency

**Comment by mareknovotny** [comment]
outdated PR, so closing now

**Pull request closed without merging**